wetmap/src/components/contexts/searchTextContext.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { createContext, useState } from 'react';
+
+type SearchTextContextType = {
+  textvalue:    string
+  setTextValue: React.Dispatch<React.SetStateAction<string>>
+};
+
+const SearchTextContextState = {
+  textvalue:     '',
+  setTextValue: () => {},
+};
```

Comment:
An interesting decision to define the base search context state here, I was curious as to why you explicitly defined an object for the base state of the search text context just to use it once in the entire codebase. Perhaps it'd be a bit more readable to simply define the original state of the SearchTextContext within the creation of the context?
```suggestion

```

Reply:
Maybe I'm not understanding if there are other motives for this? Let me know if you disagree.

Reply:
I see this same thing is done multiple times for the other contexts. Can you let me know why you've chosen to define these states? Are SearchTextContextState and other such defined states ever going to be reused with different values? I was under the assumption that that's what the types were for - defining new objects that fit the context state type.

Reply:
the types have to be defined here, otherwise typescript complains that it doesn't know what type they are on the import statements, which is why we are seeing it underline them all on pages like the diveShop and diveSite pages